src/backend/utils/mmgr/memprot.c

```diff
@@ -245,26 +265,17 @@ GPMemoryProtect_TrackStartupMemory(void)
 	 */
 	bytes += 6L << BITS_IN_MB;
 
-#ifdef USE_ORCA
-	/* When compile with ORCA it will commit 6MB more */
-	bytes += 6L << BITS_IN_MB;
-
-	/*
-	 * When optimizer_use_gpdb_allocators is on, at least 2MB of above will be
-	 * tracked by vmem tracker later, so do not recount them.  This GUC is not
-	 * available until gpdb 5.1.0 .
-	 */
-#if GP_VERSION_NUM >= 50100
-	if (optimizer_use_gpdb_allocators)
-		bytes -= 2L << BITS_IN_MB;
-#endif  /* GP_VERSION_NUM */
-#endif  /* USE_ORCA */
-
 	/* Leave some buffer for extensions like metrics_collector */
 	bytes += 2L << BITS_IN_MB;
 
+	/* freeze the addin request size and include it */
+	startup_mem_addin_request_allowed = false;
+	bytes = add_size(bytes, startup_mem_total_addin_request);
+
+	/* Ensure we do not overflow when converting to signed */
+	Assert((int64)bytes > 0);
```

Comment:
Do we really need this assertion here? The add_size function checks the result for overflow.